### ISSUE TITLE
InventoryAction#MOVE_TO_OTHER_INVENTORY not being used on Bedrock clients

### ISSUE THREAD

**Issue by ThexXTURBOXx** [GeyserMC/Geyser#953]
**Describe the bug**

``InventoryAction#MOVE_TO_OTHER_INVENTORY`` seems not to be used properly. When a "Java player" does shift-click on a item in his inventory, the ``InventoryAction#MOVE_TO_OTHER_INVENTORY`` action is properly used in the corresponding events (tried with ``InventoryClickEvent``). When a "Bedrock player" does the same, the event is called two times, once with ``PICKUP_ALL``, once with ``PLACE_ALL`` as actions.

**To Reproduce**
1. Open a chest on the Bedrock Client, which has joined a Spigot Server with Geyser installed
2. Shift+left click any item

**Expected behavior**

E.g. the ``InventoryClickEvent`` should be called with ``InventoryAction#MOVE_TO_OTHER_INVENTORY`` as action.

**Current behavior**
 but it gets called once with ```InventoryAction#PICKUP_ALL`` and once with ```InventoryAction#PLACE_ALL`` as Actions.

**Logs**

```
// Java Player shift+left clicks item
[20:22:18 INFO]: MOVE_TO_OTHER_INVENTORY: class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer, class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer
// Bedrock Player shift+left clicks item
[20:22:21 INFO]: PICKUP_ALL: class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer, class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer
[20:22:21 INFO]: PLACE_ALL: class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer, class org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer
```

**Server Version**

git-Paper-91 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

**Geyser Version**

Geyser build nr. 269 ab116dcbc3c50a0f91f10c9b801bfa0949306040
Floodgate build nr. 24

**Minecraft: Bedrock Edition Version**

1.16.102.0

**Additional Context**

I am not sure, if this is a limitation of the bedrock protocol. Also, this could relate partly to #763.

**Comment by bundabrg** [comment]
This may be a bedrock implementation issue. I'm not entirely sure we can reliably know that a shift click has occurred with the current client side inventory system. All we get is that a bunch of items that was in slot 53 is now in slot 8. We can infer that perhaps a shift-click was used and pass that to the java side but should a player actually have picked up all and drop all into the correct slots the packet will likely look the same.

I've not looked at ServerAuthoritative inventories yet so I'm not sure if that will provide more info on how things were moved.

**Comment by ThexXTURBOXx** [comment]
Okay, I already thought, this might be the case.
I have noticed that this breaks some of my spigot plugins and therefore I will need to find a solution.
Thrn I think, I will have to add a little bit more functionality in my EventHandlers for Bedrock users.
Thank you very much, though!

**Comment by RednedEpic** [comment]
As mentioned above, this appears to be a limitation in the current inventory system we use. Closing for now, but assuming something like this is still not implemented once the inventory rewrite is done, feel free to add a comment here or open up a new issue :).
